iofog-agent-daemon/src/test/java/org/eclipse/iofog/diagnostics/ImageDownloadManagerTest.java

```diff
@@ -0,0 +1,223 @@
+
+/**
+ * Agent Exception
+ * @author nehanaithani
+ *
+ */
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ImageDownloadManager.class, DockerUtil.class, DockerClientBuilder.class, DockerClient.class,
+        LoggingService.class, Orchestrator.class, DefaultDockerClientConfig.class,
+        Configuration.class, Container.class, CommandShellExecutor.class})
+public class ImageDownloadManagerTest {
+    private ImageDownloadManager imageDownloadManager;
+    private Orchestrator orchestrator;
+    private DockerUtil dockerUtil;
+    private String microserviceUuid;
+    private DockerClient dockerClient;
+    private DefaultDockerClientConfig defaultDockerClientConfig;
+    private DockerClientBuilder dockerClientBuilder;
+    private Container container;
+    private CommandShellResultSet<List<String>, List<String>> resultSetWithPath;
+    private List<String> error;
+    private List<String> value;
+
+    @Before
+    public void setUp() throws Exception {
+        microserviceUuid = "microservice-id";
+
+        imageDownloadManager = mock(ImageDownloadManager.class);
+
+        mockStatic(Configuration.class);
+        when(Configuration.getDockerUrl()).thenReturn("unix://dockerUrl/");
+        when(Configuration.getDockerApiVersion()).thenReturn("19.03.1");
+
+        orchestrator = mock(Orchestrator.class);
+
+        defaultDockerClientConfig = mock( DefaultDockerClientConfig.class);
+        dockerClientBuilder = mock( DockerClientBuilder.class);
+        mockStatic(DockerClientBuilder.class);
+        dockerClient = mock(DockerClient.class);
+        mockStatic(DockerClient.class);
+
+        mockStatic(CommandShellExecutor.class);
+
+        when(DockerClientBuilder.getInstance(any(DefaultDockerClientConfig.class))).thenReturn(dockerClientBuilder);
+        when( dockerClientBuilder.build()).thenReturn(dockerClient);
+
+        dockerUtil = mock(DockerUtil.class);
+        mockStatic(DockerUtil.class);
+
+        container = mock(Container.class);
+
+        when(DockerUtil.getInstance()).thenReturn(dockerUtil);
+
+
+        doAnswer(new Answer() {
+            @Override
+            public Object answer(InvocationOnMock invocation) throws Throwable {
+                return null;
+            }
+        }).
+                when(orchestrator).sendFileToController(any(), any());
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        error = null;
+        value =null;
+        resultSetWithPath = null;
+    }
+
+    /**
+     * When DockerUtil.getInstance().getContainer(microserviceUuid) returns null
+     * then createImageSnapshot returns
+     */
+    @Test
+    public void createImageSnapshotWithNullContainer() {
+        when(dockerUtil.getContainer(microserviceUuid)).thenReturn(null);
+
+        imageDownloadManager.createImageSnapshot(orchestrator, microserviceUuid);
+        verify(dockerUtil, atLeastOnce()).getContainer(microserviceUuid);
+    }
```

Comment:
Also verify it logs `Image snapshot: container not running.`

Reply:
updated